### DISCUSSION TITLE
referentialIntegrity bug in 4.7.0

### DISCUSSION THREAD

**Discussion by dansteryoo** [prisma/prisma#16559] [Q&A] [answered]
upgraded my CLI and getting this error and no idea what it means.
```

{
  error: PrismaClientInitializationError: 
  Invalid `prisma.blogPost.findMany()` invocation 
  
    68   publicOnly,
    69  });
    70
  → 71 const blogPosts = await prisma.blogPost.findMany(
  error: Error validating datasource `db`: 
  This option can only be set if the preview feature is enabled in a generator block.
  
  Example:
  
  generator client {
      provider = "prisma-client-js"
      previewFeatures = ["referentialIntegrity"]
  }
  
    -->  schema.prisma:12
     | 
  11 |   url          = env("DATABASE_URL")
  12 |   relationMode = "prisma"
  13 | }
     | 
  
```

Screenshot of my schema: 

<img width="648" alt="Screenshot 2022-11-30 at 12 57 13 PM" src="https://user-images.githubusercontent.com/55809283/204874423-e96c29e7-11d8-4fe8-b10a-6bd5ada1a9af.png">

**Comment by janpio** [comment, marked as answer]
Are you sure you updated your Prisma Client to 4.7.0? 
What is the `prisma generate` output? What about `prisma -v?

**Reply by dansteryoo** [reply]
Hey thanks for the quick reply. 

Please see screenshot below: 

![Screenshot 2022-11-30 at 4 25 50 PM](https://user-images.githubusercontent.com/55809283/204911427-9516ac42-7af8-415d-b7ac-fd166fbbde65.png)

**Reply by dansteryoo** [reply]
For a little more context.. this happens when I’m trying to pre-render/SSR from my NextJS frontend blog pages.. otherwise, it works fine.

**Reply by dansteryoo** [reply]
here's the full error: 

```{
  error: PrismaClientInitializationError: 
  Invalid `prisma.blogPost.findMany()` invocation in
  /Users/pd/Projects/faithlyapp-backend/src/graphql/resolvers/blogPost.ts:71:49
  
    68   publicOnly,
    69   isFaithlyAdmin,
    70 });
  → 71 const blogPosts = await prisma.blogPost.findMany(
  error: Error validating datasource `db`: 
  This option can only be set if the preview feature is enabled in a generator block.
  
  Example:
  
  generator client {
      provider = "prisma-client-js"
      previewFeatures = ["referentialIntegrity"]
  }
  
    -->  schema.prisma:12
     | 
  11 |   url          = env("DATABASE_URL")
  12 |   relationMode = "prisma"
  13 | }
     | 
  
  Validation Error Count: 1
      at RequestHandler.handleRequestError (/Users/pd/Projects/faithlyapp-backend/node_modules/@prisma/client/runtime/index.js:34857:13)
      at RequestHandler.handleAndLogRequestError (/Users/pd/Projects/faithlyapp-backend/node_modules/@prisma/client/runtime/index.js:34815:12)
      at /Users/pd/Projects/faithlyapp-backend/node_modules/@prisma/client/runtime/index.js:35361:25
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async PrismaClient._executeRequest (/Users/pd/Projects/faithlyapp-backend/node_modules/@prisma/client/runtime/index.js:35929:22)
      at async PrismaClient._request (/Users/pd/Projects/faithlyapp-backend/node_modules/@prisma/client/runtime/index.js:35900:16) {
    clientVersion: '4.7.0',
    errorCode: undefined
  }
}```

**Reply by dansteryoo** [reply]
HEY! 

So it's just magically working now lol. Dunno if you guys did anything in the back end. 

Or maybe it's just a Demogorgon thing? 

In either case, appreciate the follow-up. Keep up the good work!

Grateful for all that you do. 

Thanks!

**Reply by janpio** [reply]
We don't have a backend thing (unless you use the Data Proxy via Prisma Data Platform), so nothing to jiggle there :D 

I guess there was a cached version of an older Client somehow that still required the preview feature to be set.

**Reply by dansteryoo** [reply]
oh good call.. maybe it was a caching issue. thanks!!!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
